lua/telescope/previewers/buffer_previewer.lua

```diff
@@ -15,6 +18,47 @@ local previewers = {}
 
 local ns_previewer = vim.api.nvim_create_namespace "telescope.previewers"
 
+local function set_timeout_message(bufnr, winid)
+  local height = vim.api.nvim_win_get_height(winid)
+  local width = vim.api.nvim_win_get_width(winid)
+  vim.api.nvim_buf_set_lines(
+    bufnr,
+    0,
+    -1,
+    false,
+    utils.repeated_table(height, table.concat(utils.repeated_table(width, "╱"), ""))
+  )
+  local anon_ns = vim.api.nvim_create_namespace ""
+  local lines = { "                       ", "  Previewer timed out  ", "                       " }
+  local col = math.floor((width - strings.strdisplaywidth(lines[2])) / 2)
+  for i, line in ipairs(lines) do
+    vim.api.nvim_buf_set_extmark(
+      bufnr,
+      anon_ns,
+      math.floor(height / 2) - 1 + i,
+      0,
+      { virt_text = { { line, "Normal" } }, virt_text_pos = "overlay", virt_text_win_col = col }
+    )
+  end
+end
+
+-- modified vim.split to incorporate a timer
+local function split(s, sep, plain, opts)
+  opts = opts or {}
+  local t = {}
+  for c in vim.gsplit(s, sep, plain) do
+    table.insert(t, c)
```

Comment:
I was just about to comment lets not use table.insert but thats a lie too. I have been brainwashed for so long.

```lua
local fd = assert(vim.loop.fs_open("files", "r", 438))
local stat = assert(vim.loop.fs_fstat(fd))
local data = assert(vim.loop.fs_read(fd, stat.size, 0))
assert(vim.loop.fs_close(fd))

local bench = require "plenary.benchmark"

bench("table.insert vs index", {
  warmup = 10,
  runs = 80,
  fun = {
    {
      "table.insert",
      function()
        local result = {}
        for line in vim.gsplit(data, "\n") do
          table.insert(result, line)
        end
        return result
      end,
    },
    {
      "index",
      function()
        local result = {}
        local i = 1
        for line in vim.gsplit(data, "\n") do
          result[i] = line
          i = i + 1
        end
        return result
      end,
    },
  },
})
```
![image](https://user-images.githubusercontent.com/15233006/133140882-63e92f07-bbb6-42ae-ad28-5ede347c71b4.png)

Reply:
Okay index is slower for big files(250k), faster for small files (100)

Reply:
Noticeably so that we should change?

Reply:
8.1 μs (index) vs 18.0 μs (table.insert) for a 100 line file